vignettes/projecting_incidence.Rmd

```diff
@@ -0,0 +1,265 @@
+1. There is no reporting delay
+1. Reporting rate is constant through the course of the epidemic
+1. No interventions have been implemented
+1. Population is homogeneous and well-mixed
+
+```{r simulations, message=FALSE}
+set.seed(1234)
+
+
+sim_chain_sizes <- lapply(
+  seq_len(sim_rep),
+  function(sim) {
+    chain_sim(
+      n = length(start_times),
+      offspring = "nbinom",
+      mu = 2.5,
+      size = 0.58,
+      stat = "size",
+      infinite = chain_threshold,
+      serial = serial_interval,
+      t0 = start_times,
+      tf = projection_end_day,
+      tree = TRUE
+    ) %>%
+      mutate(sim = sim)
+  }
+)
+
+sim_output <- do.call(rbind, sim_chain_sizes)
+
+head(sim_output)
+```
+
+From the simulated data, we count the median daily cases across 
+all simulations and overlay that over a plot of all the projections through time.
+
+#### Post-processing
+
+```{r post_processing}
+index_date <- min(covid19_sa$date)
+
+# Daily number of cases for each simulation
+incidence_ts <- sim_output %>%
+  mutate(day = ceiling(time)) %>%
+  group_by(sim, day) %>%
+  summarise(cases = n()) %>%
+  ungroup()
+
+# Add dates
+incidence_ts <- incidence_ts %>%
+  group_by(sim) %>%
+  mutate(date = index_date + (0:(n() - 1))) %>%
+  ungroup()
+
+## Median daily number of cases aggregated across all simulations
+median_daily_cases <- incidence_ts %>%
+  group_by(day) %>%
+  summarise(median_cases = median(cases)) %>%
+  ungroup() %>%
+  arrange(day)
+
+# Add dates
+median_daily_cases <- median_daily_cases %>%
+  mutate(date = index_date + 0:projection_end_day) %>%
+  ungroup()
+
+```
+
+
+### Visualization
+
+```{r viz, fig.cap ="COVID-19 incidence projected over a two week window. The gray lines represent individual simulations, red connected dots represent the median daily cases across all simulations, and the black triangles represent the observed data.", fig.width=2.0, fig.height=1.8}
+# Visualization
+cases_plot <- ggplot(data = incidence_ts) +
+  geom_line(aes(
+    x = date,
+    y = cases,
+    group = sim
+  ),
+  color = "grey",
+  linewidth = 1.2,
+  alpha = 0.25
+  ) +
+  geom_point(
+    data = median_daily_cases,
+    aes(
+      x = date,
+      y = median_cases
+    ),
+    color = "tomato3",
+    size = 0.75
+  ) +
+  geom_line(
+    data = median_daily_cases,
+    aes(
+      x = date,
+      y = median_cases
+    ),
+    color = "tomato3",
+    linewidth = 0.25
+  ) +
+  geom_point(
+    data = covid19_sa,
+    aes(
+      x = date,
+      y = cases
+    ),
+    color = "black",
+    size = 0.25,
+    shape = 24
+  ) +
+  scale_x_continuous(
+    breaks = seq(min(incidence_ts$date), max(incidence_ts$date), 10),
+    labels = seq(min(incidence_ts$date), max(incidence_ts$date), 10)
+  ) +
+  scale_y_continuous(
+    breaks = seq(0, max(incidence_ts$cases) + 200, 250),
+    labels = seq(0, max(incidence_ts$cases) + 200, 250)
+  ) +
+  labs(x = "Date", y = "Daily cases (median)") +
+  theme_minimal(base_size = 4) +
+  NULL
```

Comment:
Why this `+ NULL`? Suggest removing, is not necessary for the plot unless there's something I don't understand.

Reply:
It's useful for debugging but you're right that it probably isn't necessary here. When you add NULL to the end, it makes it easier to remove layers without worrying about trailing +'s.